src/game/client/water/w3dwatertracks.cpp

```diff
@@ -325,9 +329,204 @@ int WaterTracksObj::Update(int ms_elapsed)
 
 void Test_Water_Update()
 {
-    // todo needs InGameUI
-#ifdef GAME_DLL
-    Call_Function<void>(PICK_ADDRESS(0x007A22C0, 0x0064DC2D));
+#ifdef BUILD_WITH_D3D8
+    static bool pause_waves;
+    static POINT mouse_anchor;
+    static Coord3D terrain_point_start;
+    static Coord3D terrain_point_end;
+    static WaterTracksObj *track;
+    static WaterTracksObj *track2;
+    static int track_edit_mode;
+    static int have_start;
+    static int have_end;
+    static int do_init = 1;
+    static WaveType current_wave_type = WAVE_TYPE_OCEAN;
+    static int track_edit_mode_reset = 1;
+    static int add_point_reset = 1;
+    static int delete_track_reset = 1;
+    static int save_tracks_reset = 1;
+    static int load_tracks_reset = 1;
+    static int change_type_reset = 1;

```

Comment:
It looks like some of these could be bool, because they only are used with 0 and 1 values.

Reply:
fixed.